rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/telemetry/TbMsgAttributesNode.java

```diff
@@ -154,11 +154,27 @@ private boolean checkNotifyDeviceMdValue(String notifyDeviceMdValue) {
         return StringUtils.isEmpty(notifyDeviceMdValue) || Boolean.parseBoolean(notifyDeviceMdValue);
     }
 
-    private AttributeScope getScope(String mdScopeValue) {
+    private AttributeScope getScope(TbMsg msg) {
+        if (config.isUseAttributesScopeTemplate()) {
+            try {
+                return AttributeScope.parseFrom(TbNodeUtils.processPattern(config.getScope(), msg));
+            } catch (Exception e) {
+                String mdScopeValue = msg.getMetaData().getValue(SCOPE);
```

Comment:
Consider just adding a hint that scope can be set via metadata field. Customers than will do the needed transformations on their own.